### PR TITLE
Use genuine BN type

### LINKED ISSUE #2350: [types][web3-utils] Use bn.js directly for class BN instead of wrapper
#### Expected behavior
Let `BN` denote the class as exported from `web3-utils`, not `bn.js`.
Assigning results of `BN` methods, like `add()`, to variables of type `BN` should work.

#### Actual behavior
Doesn't compile, throws typescript errors.

#### Steps to reproduce the behavior with error logs
The code 
```ts
import { BN } from "web3-utils";
let a = toBN(0); // or new BN(0)
let b: BN;
b = a.add(toBN(1));
```
gives error
```
error TS2741: Property 'super' is missing in type 'BN' but required in type 'BN'.

71 b = a.add(toBN(1));
   ~

  node_modules/web3-utils/types/index.d.ts:73:5
    73     super(number: number | string | number[] | Buffer | BN, base?: number | 'hex', endian?: 'le' | 'be'): BigNumber;
           ~~~~~
    'super' is declared here.
```
So right now I do arithmetics with `a.add(toBN(1)) as BN` hacks.

Now I could instead do `import BN = require('bn.js');` but then I get another error when passing those `BN`s into, e.g., `web3.utils.soliditySha3`:
```
error TS2345: Argument of type 'BN' is not assignable to parameter of type 'Mixed'.
  Property 'super' is missing in type 'BN' but required in type 'BN'.
```

#### Background
The web3-utils type declarations internally first do
```ts
import BigNumber = require('bn.js');
```
(which in itself is confusing, because `bn.js` is usually imported as `BN` and `BigNumber` is used for `bignumber.js`) and then declares and exports a wrapper class
```ts
export class BN extends BigNumber {
    constructor(number: number | string | number[] | Buffer | BN, base?: number | 'hex', endian?: 'le' | 'be');
    super(number: number | string | number[] | Buffer | BN, base?: number | 'hex', endian?: 'le' | 'be'): BigNumber;
}
```
Why is that? Why doesn't web3 just use `bn.js` directly?

This whole `BN` situation is a bit confusing, see also  #2171
 
#### Versions
Tested with node `v11.9.0` web3 version `1.0.0-beta.43` and typescript `3.3.1` in a truffle `v5.0.2` project.

### PR DESCRIPTION
**Pull request by vshab**
## Description

Don't wrap BN but use type from BN library to improve compatibility with BN operations. Without this change this code is invalid:
```ts
const bigNumber = new BN(3);

fromWei(bigNumber.add(new BN(100)));
```

Fixes #2350

## Type of change

- [x] Bug fix 

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no warnings.
- [x] I have updated or added types for all modules I've changed
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` in the root folder with success and extended the tests if necessary.
- [x] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [x] I ran ```npm run dtslint``` in the root folder and tested that all my types are correct
- [x] I have tested my code on an ethereum test network.

### CODE REVIEW

File: packages/web3-core/types/index.d.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@
  */
 
 import * as net from 'net';
+import BN = require('bn.js');
```
Comment (nivida) [inline packages/web3-core/types/index.d.ts:21]: Could you change it to: ``import BN from 'bn.js'``

File: packages/web3-utils/types/index.d.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@
  * @date 2018
  */
 
-import BigNumber = require('bn.js');
+import BN = require('bn.js');
```
Comment (nivida) [inline packages/web3-utils/types/index.d.ts:23]: Could you change it to: ``import BN from 'bn.js'``

Comment (nivida) [reply packages/web3-utils/types/index.d.ts:23]: Could you do this in all files? :)

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-core/types/index.d.ts (hunk 1 above)
Comment (vshab) [reply packages/web3-core/types/index.d.ts:21]: Hello!

I can do it, no problem. But according to TypeScript [handbook](https://www.typescriptlang.org/docs/handbook/modules.html) 
>When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module.

BN is exported using `export =`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bn.js/index.d.ts#L584

What do you think?

Comment (nivida) [reply packages/web3-core/types/index.d.ts:21]: Great, thanks for referencing the documentation! Yes, then we will use your way ;-)

**Review by nivida — APPROVED** [review]

**Comment by vshab** [comment]
@nivida CI fixed

**Pull request merged**